Dockerfile

```diff
@@ -79,22 +69,27 @@ RUN git clone https://github.com/aiidalab/jupyterlab-theme && \
      jupyter labextension install *.tgz && \
     cd ..
 
-# populate reentry cache for root user https://pypi.python.org/pypi/reentry/
+# Populate reentry cache for root user https://pypi.python.org/pypi/reentry/.
 RUN reentry scan
 
 # Prepare user's folders for AiiDA lab launch.
 COPY opt/aiidalab-singleuser /opt/
 COPY opt/prepare-aiidalab.sh /opt/
 COPY my_init.d/prepare-aiidalab.sh /etc/my_init.d/80_prepare-aiidalab.sh
 
-# Start Jupyter notebook
+# Start Jupyter notebook.
 COPY opt/start-notebook.sh /opt/
 COPY service/jupyter-notebook /etc/service/jupyter-notebook/run
 
+# Install some useful packages that are not available on PyPi
+RUN conda install --yes -c conda-forge rdkit
+RUN conda install --yes -c openbabel openbabel "tornado<5"
```

Comment:
what happens if you don't put "tornado <5" here? Does it not see the tornado version that is already installed?

Also, when providing "tornado<5", does it install a second version of tornado over the pip one?
Just curious to understand what is going on here...

Reply:
> what happens if you don't put "tornado <5" here? Does it not see the tornado version that is already installed?

Just tested: conda will see that currently installed tornado (that is 4.5.3) is still compatible with openbabel dependencies and won't touch tornado. So I will remove "tornado<5" then.

> Also, when providing "tornado<5", does it install a second version of tornado over the pip one?

No, looks like conda and PyPI packages both get installed in the same folder which is `/opt/conda/lib/python3.7/site-packages`. Just tested it with `signac` package:

```
root@cc607f96dcd2:/opt/jupyterlab-theme# pip install signac
Collecting signac
  Downloading signac-1.3.0-py3-none-any.whl (181 kB)
     |████████████████████████████████| 181 kB 2.4 MB/s
Collecting filelock~=3.0
  Downloading filelock-3.0.12-py3-none-any.whl (7.6 kB)
Collecting deprecation>=2
  Downloading deprecation-2.0.7-py2.py3-none-any.whl (11 kB)
Requirement already satisfied: packaging>=15.0 in /opt/conda/lib/python3.7/site-packages (from signac) (19.2)
Requirement already satisfied: pyparsing>=2.0.2 in /opt/conda/lib/python3.7/site-packages (from packaging>=15.0->signac) (2.4.6)
Requirement already satisfied: six in /opt/conda/lib/python3.7/site-packages (from packaging>=15.0->signac) (1.14.0)
Installing collected packages: filelock, deprecation, signac
Successfully installed deprecation-2.0.7 filelock-3.0.12 signac-1.3.0


root@cc607f96dcd2:/opt/jupyterlab-theme# pip freeze | grep signac
signac==1.3.0

root@cc607f96dcd2:/opt/jupyterlab-theme# conda list | grep signac
signac                    1.3.0                    pypi_0    pypi

In [1]: import signac

In [2]: signac.__file__
Out[2]: '/opt/conda/lib/python3.7/site-packages/signac/__init__.py'

```